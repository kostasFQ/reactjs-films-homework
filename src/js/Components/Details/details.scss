.container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-left: 1%;
}

.h1 {
  width: 100%;
  margin: 0;
  padding: 0;
  font-size: 3vw;
  text-align: start;
  text-shadow: 0 0 5px black;

  @media only screen and (max-width: 600px) {
    font-size: 4vw;
  }
}
