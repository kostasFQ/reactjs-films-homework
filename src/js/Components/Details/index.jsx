import React from 'react';
import PropTypes from 'prop-types';
import styles from './details.scss';
import Rating from '../Rating';
import Genres from '../Genres';

const Details = (props) => {
  const {
    original_title, genre_ids, vote_average,
  } = props;

  return (
    <div className={styles.container}>
      <h1 className={styles.h1}>{original_title.toUpperCase()}</h1>
      <Genres genre_ids={genre_ids} />
      <Rating rating={vote_average} />
    </div>
  );
};

export default Details;

Details.propTypes = {
  original_title: PropTypes.string.isRequired,
  vote_average: PropTypes.number.isRequired,
  genre_ids: PropTypes.arrayOf(PropTypes.number).isRequired,
};
