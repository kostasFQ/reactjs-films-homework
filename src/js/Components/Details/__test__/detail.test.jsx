import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../index';

const props = {
  original_title: 'title',
  genre_ids: [12, 16, 18],
  vote_average: 4.5,
};

test('test component', () => {
  const render = renderer.create(
    <Details {...props} />,
  );

  const component = render.toJSON();
  expect(component).toMatchSnapshot();
});
