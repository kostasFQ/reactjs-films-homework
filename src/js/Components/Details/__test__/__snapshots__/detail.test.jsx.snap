// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component 1`] = `
<div
  className="container"
>
  <h1
    className="h1"
  >
    TITLE
  </h1>
  <ul
    className="list"
  >
    <li
      className="item"
    >
      Adventure
    </li>
    <li
      className="item"
    >
      Animation
    </li>
    <li
      className="item"
    >
      Drama
    </li>
  </ul>
  <div
    className="container"
  >
    <div
      className="ratingBox"
    >
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
    </div>
    <div
      className="x"
      style={
        Object {
          "width": "90px",
        }
      }
    >
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
      <img
        alt="star"
        className="star"
        src="test-file-stub"
      />
    </div>
    <div
      className="container"
    >
      4.5
    </div>
  </div>
</div>
`;
